dev-tools/mage/dockerbuilder.go

```diff
@@ -103,8 +106,35 @@ func (b *dockerBuilder) exposePorts() []string {
 
 func (b *dockerBuilder) copyFiles() error {
 	for _, f := range b.Files {
+		source := f.Source
+		var checkFn func(string) bool
 		target := filepath.Join(b.beatDir, f.Target)
-		if err := Copy(f.Source, target); err != nil {
+
+		if f.ExpandSpec {
+			specFilename := filepath.Base(source)
+			specContent, err := os.ReadFile(source)
+			if err != nil {
+				if os.IsNotExist(err) {
+					return nil
+				}
+				return fmt.Errorf("failed reading spec file for component %q: %w", specFilename, err)
+			}
+
+			// create filter
+			allowedPaths, err := component.ParseComponentFiles(specContent, specFilename, true)
+			if err != nil {
+				return fmt.Errorf("failed computing component files %q: %w", specFilename, err)
+			}
+			checkFn, err = install.SkipComponentsPathWithSubpathsFn(allowedPaths)
```

Comment:
We are introducing a dependency here between `dev-tools` and `elastic-agent` install package here.
Same with the `component.ParseComponentFiles` call a few lines above this.
I am not sure that this is the correct dependency direction between build/package code and the code being packaged

Reply:
we do have flavors so besides duplicating logic there needs to be some kind of dependency.

Reply:
We also have a declarative YAML where package specs can be configured and that would not introduce a dev-tool -> elastic-agent code dependency.

Since this is under time constraints to get in 9.0, I guess we can merge it as-is and rework it later